cpp/ql/src/Telemetry/Diagnostics.qll

```diff
@@ -0,0 +1,32 @@
+import cpp
+
+/**
+ * A syntax error.
+ */
+class SyntaxError extends CompilerError {
+  SyntaxError() { this.getTag().matches("exp_%") }
+}
+
+/**
+ * A cannot open file error.
+ * Typically this is due to a missing include.
+ */
+class CannotOpenFile extends CompilerError {
+  CannotOpenFile() { this.hasTag("cannot_open_file") }
```

Comment:
Shouldn't this also cover `cannot_open_file_reason`?

Reply:
I will have a look at these tags 👍